activesupport/lib/active_support/message_verifier.rb

```diff
@@ -53,7 +53,9 @@ def valid_message?(signed_message)
       return if signed_message.nil? || !signed_message.valid_encoding? || signed_message.blank?
 
       data, digest = signed_message.split("--".freeze)
-      data.present? && digest.present? && ActiveSupport::SecurityUtils.secure_compare(digest, generate_digest(data))
+      data.present? && digest.present? && ActiveSupport::SecurityUtils.fixed_length_secure_compare(digest, generate_digest(data))
+    rescue ArgumentError
+      false
```

Comment:
@jeremy this is the only intrusive change. This is so we guarantee return of true/false values, that others depend on, like cookies.
I think its ok to do this because:
- `ArgumentError` raise is only on negative cases.
- On positive cases, we won't need to do extra SHA over digest, which out-weighs negative cases.